File: packages/people/android/src/main/java/io/capkit/people/config/PeopleConfig.kt
package io.capkit.people.config

import com.getcapacitor.Plugin
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json

/**
 * Data Transfer Object (DTO) for parsing the People configuration
 * section in capacitor.config.ts.
 *
 * The keys mirror the TypeScript `PeopleConfig` interface
 * (packages/people/src/definitions.ts).
 */
@Serializable
private data class RawPeopleConfig(
  @SerialName("verboseLogging")
  val verboseLogging: Boolean = false,
)

/**
 * Plugin configuration holder for the People plugin.
 *
 * This class is responsible for reading and parsing static configuration values
 * defined under the `plugins.People` key in `capacitor.config.ts`.
 *
 * Architectural rules:
 * - Read once during plugin initialization in the load() phase.
 * - Configuration values are read-only at runtime.
 * - Consumed only by native code.
 */
class PeopleConfig(
  plugin: Plugin,
) {
  // -----------------------------------------------------------------------------
  // Properties
  // -----------------------------------------------------------------------------

  /**
   * Enables verbose native logging via PeopleLogger.
   *
   * When true, additional debug information and lifecycle events are printed to Logcat.
   * This setting is read-only and applied during plugin initialization.
   *
   * Default: false
   */
  val verboseLogging: Boolean

  // -----------------------------------------------------------------------------
  // Initialization
  // -----------------------------------------------------------------------------

  init {
    val configObject = plugin.config.getConfigJSON()
    val parsedConfig = parseConfig(configObject?.toString())

    verboseLogging = parsedConfig.verboseLogging
  }

  private companion object {
    private val json =
      Json {
        ignoreUnknownKeys = true
        isLenient = true
      }

    private fun parseConfig(jsonString: String?): RawPeopleConfig {
      if (jsonString.isNullOrBlank()) return RawPeopleConfig()
      return try {
        json.decodeFromString<RawPeopleConfig>(jsonString)
      } catch (_: Exception) {
        RawPeopleConfig()
      }
    }
  }
}

File: packages/people/ios/Sources/PeoplePlugin/config/PeopleConfig.swift
import Foundation
import Capacitor

/**
 * Plugin configuration container for the People plugin.
 *
 * This struct is responsible for reading and exposing static configuration
 * values defined under the `People` key in `capacitor.config.ts`.
 *
 * Architectural rules:
 * - Read once during plugin initialization (`load()`).
 * - Treated as immutable runtime input.
 * - Consumed only by native code (never by JavaScript).
 */
public struct PeopleConfig {

    // MARK: - Configuration Keys

    /**
     * Internal structure to maintain consistent configuration keys.
     */
    private struct Keys {
        static let verboseLogging = "verboseLogging"
    }

    // MARK: - Public Properties

    /**
     * Enables verbose native logging via PeopleLogger.
     *
     * When enabled, additional debug information is printed to the Xcode console.
     * Default: false
     */
    public let verboseLogging: Bool

    // MARK: - Private Defaults

    private static let defaultVerboseLogging: Bool = false

    // MARK: - Initialization

    /**
     * Initializes the configuration by reading values from the Capacitor bridge.
     *
     * - Parameter plugin: The CAPPlugin instance used to access typed configuration via `getConfig()`.
     */
    init(plugin: CAPPlugin) {
        let config = plugin.getConfig()

        // Extract verboseLogging with fallback to default
        self.verboseLogging = config.getBoolean(
            Keys.verboseLogging,
            Self.defaultVerboseLogging
        )
    }
}
